OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnMovement.cs

```diff
@@ -45,17 +51,35 @@ protected override void Created(Actor self)
 			base.Created(self);
 		}
 
-		void ITick.Tick(Actor self)
+		void UpdateCondition(Actor self, MovementType types)
 		{
 			if (conditionManager == null)
 				return;
 
-			var isMovingVertically = Info.ConsiderVerticalMovement ? movement.IsMovingVertically : false;
-			var isMoving = !IsTraitDisabled && !self.IsDead && (movement.IsMoving || isMovingVertically);
-			if (isMoving && conditionToken == ConditionManager.InvalidConditionToken)
-				conditionToken = conditionManager.GrantCondition(self, Info.Condition);
-			else if (!isMoving && conditionToken != ConditionManager.InvalidConditionToken)
+			var validMovement = !IsTraitDisabled
```

Comment:
The goal here is to say "are any of the flags in `Info.ValidMovementTypes` set in `types`", right?

This can be written as the much simpler 
```csharp
var validMovement = !IsTraitDisabled && (types & Info.ValidMovementTypes) != 0;
```

The `&` operator returns 1 for each bit that is set in both `types` AND `Info.ValidMovementTypes`. So this will only return 0 overall if all the `MovementTypes` set in `types` are not set in `Info.ValidMovementTypes`, and visa versa.